tests/integration/query/one_to_many/with_filter_test.go

```diff
@@ -17,274 +17,468 @@ import (
 					"age": 65,
 					"verified": true
 				}`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 1,
 				// bae-b769708d-f552-5c3d-a402-ccfd7ac7fb04
-				`{
+				Doc: `{
 					"name": "Cornelia Funke",
 					"age": 62,
 					"verified": false
 				}`,
 			},
-		},
-		Results: []map[string]any{
-			{
-				"name": "John Grisham",
-				"age":  uint64(65),
-				"p1": []map[string]any{
-					{
-						"name":   "Painted House",
-						"rating": 4.9,
-					},
-				},
-				"p2": []map[string]any{
+			testUtils.Request{
+				Request: `query {
+					Author {
+						name
+						age
+						p1: published(filter: {rating: {_gt: 4.6}}) {
+							name
+							rating
+						}
+						p2: published(filter: {rating: {_lt: 4.6}}) {
+							name
+							rating
+						}
+					}
+				}`,
+				Results: []map[string]any{
 					{
-						"name":   "A Time for Mercy",
-						"rating": 4.5,
+						"name": "John Grisham",
+						"age":  uint64(65),
+						"p1": []map[string]any{
+							{
+								"name":   "Painted House",
+								"rating": 4.9,
+							},
+						},
+						"p2": []map[string]any{
+							{
+								"name":   "A Time for Mercy",
+								"rating": 4.5,
+							},
+						},
 					},
-				},
-			},
-			{
-				"name": "Cornelia Funke",
-				"age":  uint64(62),
-				"p1": []map[string]any{
 					{
-						"name":   "Theif Lord",
-						"rating": 4.8,
+						"name": "Cornelia Funke",
+						"age":  uint64(62),
+						"p1": []map[string]any{
+							{
+								"name":   "Theif Lord",
+								"rating": 4.8,
+							},
+						},
+						"p2": []map[string]any{},
 					},
 				},
-				"p2": []map[string]any{},
 			},
 		},
 	}
 
-	executeTestCase(t, test)
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestQueryOneToManyWithCompoundOperatorInFilterAndRelation(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "One-to-many relation query filter with compound operator and relation",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: bookAuthorGQLSchema,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 0,
+				Doc: `{
+					"name": "Painted House",
+					"rating": 4.9,
+					"author_id": "bae-41598f0c-19bc-5da6-813b-e80f14a10df3"
+				}`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 0,
+				Doc: `{
+					"name": "A Time for Mercy",
+					"rating": 4.5,
+					"author_id": "bae-41598f0c-19bc-5da6-813b-e80f14a10df3"
+					}`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 0,
+				Doc: `{
+					"name": "Theif Lord",
+					"rating": 4.8,
+					"author_id": "bae-b769708d-f552-5c3d-a402-ccfd7ac7fb04"
+				}`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 1,
+				// bae-41598f0c-19bc-5da6-813b-e80f14a10df3
+				Doc: `{
+					"name": "John Grisham",
+					"age": 65,
+					"verified": true
+				}`,
+			},
+			testUtils.CreateDoc{
+				CollectionID: 1,
+				// bae-b769708d-f552-5c3d-a402-ccfd7ac7fb04
+				Doc: `{
+					"name": "Cornelia Funke",
+					"age": 62,
+					"verified": false
+				}`,
+			},
+			testUtils.Request{
```

Comment:
suggestion: I think these tests can be improved slightly:

At the moment you have an or of three, but you are only returning a single record each time - this means that I could 'simplify' the test, by removing two of those 3 inner conditions, leaving an or of one, and then further simplify the test to remove the or.  Neither the setup, or the results would change - this offers little protection against changing the filter, the only real reason stopping developers from simplifying the test is their ability to read and understand what the title/description says the test is trying to test.

If you change it so that changing any one of the setup, act (filter), or assert (results) steps will cause the test to fail, it should be even more robust.

The same goes for the other new integration tests, they seem to all follow this pattern.

similar suggestion: I'd also drop the `verified` field from these tests too, other than the dockeys in the setup, it has no impact on the test result and is just noise.